### ISSUE TITLE
Asset field empty when using a GraphQL Query

### ISSUE THREAD

**Issue by benediktblank** [spicywebau/craft-neo#519]
#### Description
When you query an asset field inside a Neo block, no asset information is returned, just an empty array.

#### Steps to reproduce
1. Setup a neo field with a Neo field block
2. Use an underscore in the Neo Field Block handle, e.g. block_topBanner.
3. Add an asset field to the block
4. Query the field with the block inside an entry and try to request the image field
Like this for example:
backgroundImage {
  url
}
The response is just:
"backgroundImage": [],
See full request and response below.
5. When you add the field outside of the neo block, it works fine.
6. Also, when you change the Neo Field Block handle, the yaml are updated okay but when you run "php craft project-config/apply" on another system, the handles are not updated.

#### Other information
- Neo version: 2.11.18
- Craft version: 3.7.21

Full Request:
````
{
  entries(slug: "about-us") {
    id
    title
    ... on aboutUs_aboutUs_Entry {
        backgroundImage {
            url
          }
      neoField {
        ... on neoField_block_topBanner_BlockType {
          backgroundImage {
            url
          }
        }
      }
    }
  }
}
````
Full Response:
````
{
  "data": {
    "entries": [
      {
        "id": "1153",
        "title": "About us",
        "backgroundImage": [
          {
            "url": "https://ddev-craftcms.ddev.site/assets/blog/schulung-teaser-image.png"
          }
        ],
        "neoField": [
          {
            "backgroundImage": []
          }
        ]
      }
    ]
  }
}
````

**Comment by benediktblank** [comment]
When debugging this, the issue seesm to be here:
src/gql/resolvers/elements/Asset.php

````
    public static function prepareQuery($source, array $arguments, $fieldName = null)
    {
        // If this is the beginning of a resolver chain, start fresh
        if ($source === null) {
            $query = AssetElement::find();
            // If not, get the prepared element query
        } else {
            $query = $source->$fieldName;
        }

        // If it's preloaded, it's preloaded.
        if (is_array($query)) {
            return $query;
        }
````

Code goes into "else", calls $source->backgroundImage which returns an empty array.
And that just leads to return that empty array.
Apparently that empty array was "eagerLoaded"?

**Comment by benediktblank** [comment]
I managed to find out where it's populated with the empty array:
vendor/craftcms/cms/src/services/Elements.php
````
$sourceElement->setEagerLoadedElements($plan->alias, $targetElementsForSource);
if ($plan->count) {
    $sourceElement->setEagerLoadedElementCount($plan->alias, count($targetElementsForSource));
}
````


When I change it to this it works:
````
                    if ($plan->count) {
                        $sourceElement->setEagerLoadedElements($plan->alias, $targetElementsForSource);
                    }


                    if ($plan->count) {
                        $sourceElement->setEagerLoadedElementCount($plan->alias, count($targetElementsForSource));
                    }
````
But that can't be the main issue because the asset field works outside of a Neo field perfectly fine.

**Comment by benediktblank** [comment]
It seems that Eager Loading stuff for the asset field is only done when it's inside a Neo Field Block.
When putting the asset field outside the Neo Field, it's not eager loaded and the line "$query = $source->$fieldName;" correctly returns the craft\elements\db\AssetQuery object.

**Comment by ttempleton** [comment]
Thanks for reporting that, the bug with the lack of asset field results will be fixed in the next release.

Applying project config changes that include a Neo block type handle change seems to be working for me so far. Since the original bug with GraphQL queries is resolved, I'm going to close this issue, but if you can reliably reproduce the behaviour with block type handles and applying project config changes, could you please check whether other block type settings are getting updated even if the handle isn't, and post that along with detailed steps to reproduce as a new issue.

**Comment by ttempleton** [comment]
Just released Neo 2.11.19 with the GraphQL fix.
